### PR TITLE
Improve API performance

### PR DESCRIPTION
**Pull request by beagleknight**
# What and why

I have detected a few problems with the current implementation of our API which caused a slow performance on some queries.

First, I simplify some serializers so they don't send redundant information. Then I fixed some N+1 queries using eager loading of some associations.

Finally I added a `subcategories` API endpoint because they are not embedded on the `categories` API response.
# GIF Tax

### CODE REVIEW

File: app/controllers/api/action_plans/meetings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Api::ActionPlans::MeetingsController < Api::ApplicationController
+  load_and_authorize_resource :action_plan
+
+  def index
+    meeting_ids = @action_plan.proposals
+    .includes(:meetings)
```
Comment (josepjaume) [inline app/controllers/api/action_plans/meetings_controller.rb]: Indentation xD

File: app/controllers/api/action_plans/proposals_controller.rb [hunk 2]
```diff
@@ -4,6 +4,15 @@ class Api::ActionPlans::ProposalsController < Api::ApplicationController
 
   def index
     @action_plans_proposals = @action_plan.action_plans_proposals
+    .includes(:proposal => [
+      {:author => :organization},
```
Comment (josepjaume) [inline app/controllers/api/action_plans/proposals_controller.rb]: `author: :organization`

File: app/controllers/api/action_plans_controller.rb [hunk 3]
```diff
@@ -11,7 +11,9 @@ class Api::ActionPlansController < Api::ApplicationController
   def index
     set_seed
 
-    action_plans = ActionPlan.all
+    action_plans = ActionPlan
+    .includes(:revisions)
```
Comment (josepjaume) [inline app/controllers/api/action_plans_controller.rb]: No multiple lines needed

File: app/controllers/api/proposals_controller.rb [hunk 4]
```diff
@@ -13,6 +13,11 @@ def index
 
     proposals = @current_order == "recommended" ? Recommender.new(current_user).proposals : Proposal.all
 
+    proposals = proposals
+    .includes(:comments)
```
Comment (josepjaume) [inline app/controllers/api/proposals_controller.rb]: No multiple lines needed

File: app/frontend/action_plans/action_plan_show.component.js [hunk 5]
```diff
@@ -116,8 +96,35 @@ class ActionPlanShow extends Component {
                 namespace="action_plans"
                 useServerLinks={ true }/>
 
-              <ActionPlanReviewer />
+              <ActionPlanReviewer visible={this.state.editable} />
+
+              <h2>Estat de l'actuacio</h2>
+              <p>TODO</p>
```
Comment (josepjaume) [inline app/frontend/action_plans/action_plan_show.component.js:102]: Code highlighting FAIL

File: app/serializers/proposal_serializer.rb [hunk 6]
```diff
@@ -1,40 +1,40 @@
...
   end
 
   def total_neutral_comments
-    object.comments.neutral.count
+    object.comments.select { |c| c.alignment && c.alignment == 0 }.count
   end
 
   def voted
-    scope && scope.current_user && scope.current_user.voted_up_on?(object)
+    object.votes_for.detect do |v|
```
Comment (josepjaume) [inline app/serializers/proposal_serializer.rb]: what do you think about memoizing this?

**Comment by josepjaume** [comment]
Check the comments and 👍

File: app/controllers/api/action_plans/meetings_controller.rb (hunk 1 above)
Comment (beagleknight) [inline app/controllers/api/action_plans/meetings_controller.rb]: lol, I used the `.` as a `|>`

File: app/frontend/action_plans/action_plan_show.component.js (hunk 5 above)
Comment (beagleknight) [inline app/frontend/action_plans/action_plan_show.component.js:102]: lol XD, that's not even finished! We need to talk about the other PR @josepjaume

**Pull request merged**